ddl/ddl_worker.go

```diff
@@ -522,13 +523,20 @@ func (w *worker) runDDLJob(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64,
 	if err != nil {
 		// If job is not cancelled, we should log this error.
 		if job.State != model.JobStateCancelled {
-			log.Errorf("[ddl-%s] run DDL job err %v", w, errors.ErrorStack(err))
+			log.Errorf("[ddl-%s] run DDL job err %v, job query %s ", w, errors.ErrorStack(err), job.Query)
 		} else {
-			log.Infof("[ddl-%s] the DDL job is normal to cancel because %v", w, errors.ErrorStack(err))
+			log.Infof("[ddl-%s] the DDL job is normal to cancel because %v, job query %s", w, errors.ErrorStack(err), job.Query)
+			metrics.DDLWorkerHistogram.WithLabelValues(metrics.WorkerCancelDDLJob, job.Type.String(), metrics.RetLabel(err)).Observe(time.Since(model.TSConvert2Time(job.StartTS)).Seconds())
 		}
 
 		job.Error = toTError(err)
 		job.ErrorCount++
+		if job.ErrorCount > int64(variable.GetDDLErrorRetryLimit()) && job.Type != model.ActionAddIndex {
+			log.Infof("[ddl-%s] DDL job over maximum retry count is canceled because %v, job query %s", w, errors.ErrorStack(err), job.Query)
+			job.State = model.JobStateCancelling
```

Comment:
Is there a problem with the following scenario？
If the operation of “Add column” updates to “Delete only”, then we cancel this job.

Reply:
@ciscoxll 
If the column is “drop column” updates to “delete only”， but this job occurs many errors then this job cancel successfully. Then we return an error to the client, but this column's state is "delete only". 
I think this will be a problem.

Reply:
@zimulala I will write a `rollback` in another PR.

Reply:
Fix it on #9295